src/lineup/internal/panel/PanelTab.ts

```diff
@@ -0,0 +1,134 @@
+import {SidePanel, SearchBox} from 'lineupjs';
+import {ISearchOption} from '../LineUpPanelActions';
+import {EventHandler} from 'phovea_core/src/event';
+import PanelButton, {PanelNavButton} from './PanelButton';
+
+/**
+ * Interface for the options parameter of PanelTab
+ */
+export interface IPanelTabDesc {
+
+  /**
+   * Width of the SidePanel
+   */
+  width: string;
+
+  /**
+   * CSS class for PanelNavButton of the PanelTab
+   */
+  cssClass: string;
+
+  /**
+   * Title and Text content for the PanelNavButton of the PanelTab.
+   */
+  title: string;
+
+  /**
+   * Define the sort order of the PanelNavButtons
+   */
+  order: number;
+
+  /**
+   * Show PanelNavButton in collapsed mode
+   * @default false
+   */
+  shortcut?: boolean;
+}
+
+/**
+ * Events for PanelTab for when its showed/hidden
+ */
+export class PanelTabEvents extends EventHandler {
+
+  static readonly SHOW_PANEL = 'showPanel';
+
+  static readonly HIDE_PANEL = 'hidePanel';
+
+}
+
+/**
+ * The PanelTab creates a tab component that with can be toggled through the PanelNavButton
+ */
+export class PanelTab {
+
+  readonly node: HTMLElement;
+  readonly events: PanelTabEvents;
```

Comment:
usually we don't have instance variables like this and instead the class itself (`PanelTab` in this case) inherits from `EventHandler`). at least I haven't seen a usage like this yet

@thinkh do you have usages like this in Ordino?

Reply:
You are right. Our intention was to pass an event object, so that the extension point can react to some events, and this way avoid to pass the whole tab to the extension point. Otherwise the extension point will have access to functionality it does not need or should not

Reply:
We changed the implementation according to your suggestion and integrated the events into the panel tab.